### PR TITLE
[YUNIKORN-1559] Add a resource of `pods=1` on all pods during scheduling

### PR DESCRIPTION
**Pull request by elihschiff**
### What is this PR for?
In my cluster I am seeing occasional events like this on some of my pods which causes them to get stuck.

`54m     Warning  OutOfpods           pod/tg-spark-executor-640b4349263cc74570ae3a1e-0     Node didn't have enough resource: pods, requested: 1, used: 12, capacity: 12`

From what I can tell, nodes all have a `pods` resource value on them. Because this value already exists on nodes, my PR adds a pods=1 value to add pods inside the scheduler. Then yunikorn uses the same resource scheduling logic used for memory, cpu, ect. to limit the number of pods running on a node.

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - This change only works once the core repo is updated https://github.com/apache/yunikorn-core/pull/521. Therefore that change will need to be merged first.

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1559

### How should this be tested?

I am not 100% sure how best to test this. I fixed the tests including the e2e tests. I am testing this on my kubernetes clusters and it seems to have fixed the issue but I am not 100% sure if there is anything I should be worried about.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: go.mod
Comment (elihschiff) [inline go.mod:48]: I am leaving this here for now but I will update the go.mod/go.sum files once the other PR is merged. 

DO NOT MERGE this pr until after that point

Comment (elihschiff) [reply go.mod:48]: This PR now must be merged after https://github.com/apache/yunikorn-core/pull/523 is merged which solves the core issue limiting cluster autoscaling from working.

Comment (wilfred-s) [reply go.mod:48]: please remove the go.mod and go.sum updates, the change can be committed without the update.

File: go.sum
Comment (wilfred-s) [inline go.sum:249]: see above.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: go.mod
Comment (elihschiff) [reply go.mod]: 👍 I removed the go.mod and go.sum changes

File: pkg/common/resource.go
Comment (elihschiff) [reply pkg/common/resource.go:187]: I believe I added what you requested in the most recent commit. Let me know if what I did was incorrect.

https://github.com/apache/yunikorn-k8shim/pull/558/commits/af5ed6d4202c42bae584c15dae1f50a9fd17d428

**Comment by wilfred-s** [comment]
e2e failures are due to YUNIKORN-1666, need to re-run after that change is done before we commit

**Comment by craigcondit** [comment]
Kicked off recheck as YUNIKORN-1666 is merged.

**Review by wilfred-s — APPROVED** [review]
LGTM
waiting for the build to not fail e2e compilation

**Pull request closed without merging**
